BlueprintUI/Sources/Layout/Flow.swift

```diff
@@ -0,0 +1,390 @@
+/// and via `RowAlignment`, which controls the vertical alignment of shorter elements within a row.
+///
+/// ```swift
+/// Flow(horizontalSpacing: 10, rowSpacing: 10) {
+///     OnboardingPill("Food / drink")
+///     OnboardingPill("Retail goods").flowChild(key: "aKey")
+///     OnboardingPill("Grocery / gourmet / alcohol")
+///     OnboardingPill("Beauty / wellness bookings")
+///     OnboardingPill("Healthcare services")
+///     OnboardingPill("Something else")
+/// }
+/// ```
+///
+/// Below is a diagram showing a simple example of a `Flow` layout.
+///
+/// ```
+/// ┌─────────────────────────────────────┐
+/// │ ┌─────────┐┌─────────┐┌────────────┐│
+/// │ │    1    ││    2    ││     3      ││
+/// │ └─────────┘└─────────┘└────────────┘│
+/// │ ┌───────┐┌─────┐┌───────────┐       │
+/// │ │   4   ││  5  ││     6     │       │
+/// │ └───────┘└─────┘└───────────┘       │
+/// └─────────────────────────────────────┘
+/// ```
+public struct Flow: Element {
+
+    /// How to horizontally align items in the row when there is extra space.
+    public var horizontalAlignment: HorizontalAlignment
+
+    /// How to vertically align items when there is extra space.
+    public var rowAlignment: RowAlignment
+
+    /// Space between items within a row.
+    public var horizontalSpacing: CGFloat
+
+    /// Space between rows.
+    public var rowSpacing: CGFloat
+
+    /// The child elements of the flow layout to be laid out.
+    public var children: [Child]
+
+    /// Creates a new flow layout with the provided parameters.
+    public init(
+        horizontalAlignment: HorizontalAlignment = .leading,
+        horizontalSpacing: CGFloat = 0,
+        rowAlignment: RowAlignment = .center,
+        rowSpacing: CGFloat = 0,
+        @ElementBuilder<Flow.Child> _ children: () -> [Flow.Child]
+    ) {
+        self.horizontalAlignment = horizontalAlignment
+        self.horizontalSpacing = horizontalSpacing
+        self.rowAlignment = rowAlignment
+        self.rowSpacing = rowSpacing
+        self.children = children()
+    }
+
+    // MARK: Element
+
+    public var content: ElementContent {
+        .init(
+            layout: Layout(
+                horizontalAlignment: horizontalAlignment,
+                rowAlignment: rowAlignment,
+                horizontalSpacing: horizontalSpacing,
+                rowSpacing: rowSpacing
+            )
+        ) {
+            for child in children {
+                $0.add(traits: child.traits, key: child.key, element: child.element)
+            }
+        }
+    }
+
+    public func backingViewDescription(with context: ViewDescriptionContext) -> ViewDescription? {
+        nil
+    }
+}
+
+extension Flow {
+
+    /// How to horizontally align items in the row when there is extra space.
+    public enum HorizontalAlignment: CaseIterable {
+
+        /// Items are aligned with the leading edge.
+        case leading
```

Comment:
Would we also want to support `justify` horizontal alignment?

Reply:
In the name of YAGNI, I decided to leave it off until we have a use case for it – I haven't seen any designs asking for this yet, so, didn't add it!